src/python/pants/backend/python/target_types.py

```diff
@@ -775,6 +752,8 @@ class PythonSourceTarget(Target):
         *COMMON_TARGET_FIELDS,
         InterpreterConstraintsField,
         Dependencies,
+        # TODO(#12314): Switch to compatible_resolves if relevant.
```

Comment:
These comments don't seem necessary... but if they're going to be here, it seems like they should point to #13882 instead?

Reply:
Agreed, they were written before I read your design doc ;) the last commit removes them.